File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/C/s910447421_commented.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_N 100

int ans = 100;
int balls[MAX_N][2];
int N;
int taken[MAX_N];

int compare(const void *a, const void *b) {
    int *arr1 = (int *)a;
    int *arr2 = (int *)b;
    return arr1[0] - arr2[0];
}

int min(int a, int b) {
    return a < b ? a : b;
}

void solve() {
    for (int add = 1; add < N; add++) {
        memset(taken, 0, sizeof(taken));
        int p = balls[add][0] - balls[0][0];
        int q = balls[add][1] - balls[0][1];
        ans = min(ans, check(p, q));
    }
}

int check(int p, int q) {
    int ret = 0;
    while (1) {
        int added = 0;
        int start = 0;
        int x = -1;
        int y = -1;
        for (int i = 0; i < N; i++) {
            if (!taken[i]) {
                added = 1;
                ret++;
                start = i;
                taken[i] = 1;
                x = balls[i][0] + p;
                y = balls[i][1] + q;
                break;
            }
        }
        if (!added) break;
        for (int i = start + 1; i < N; i++) {
            if (balls[i][0] == x && balls[i][1] == y) {
                taken[i] = 1;
                x += p;
                y += q;
            }
        }
    }
    return ret;
}

int main() {
    scanf("%d", &N);
    ans = N;
    for (int i = 0; i < N; i++) {
        scanf("%d %d", &balls[i][0], &balls[i][1]);
    }
    qsort(balls, N, sizeof(balls[0]), compare);
    solve();
    for (int i = 0; i < N; i++) {
        int x = balls[i][0];
        int y = balls[i][1];
        balls[i][0] = y;
        balls[i][1] = x;
    }
    qsort(balls, N, sizeof(balls[0]), compare);
    solve();
    printf("%d\n", ans);
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/inline/comment/direct/Java/Python/s910447421_commented.py
import sys
import math

class FastScanner:
    def __init__(self):
        self.br = sys.stdin
        self.st = []

    def next(self):
        while not self.st:
            try:
                self.st = self.br.readline().split()
            except:
                return None
        return self.st.pop(0)

    def nextInt(self):
        return int(self.next())

    def nextLong(self):
        return int(self.next())

    def nextDouble(self):
        return float(self.next())

class OutPut:
    def __init__(self):
        self.w = sys.stdout

    def pln(self, x):
        self.w.write(str(x) + '\n')
        self.w.flush()

    def p(self, x):
        self.w.write(str(x))
        self.w.flush()

def min(a, b):
    return a if a < b else b

def solve():
    global ans
    for add in range(1, N):
        taken = [False] * N
        p = balls[add][0] - balls[0][0]
        q = balls[add][1] - balls[0][1]
        ans = min(ans, check(p, q))

def check(p, q):
    ret = 0
    while True:
        added = False
        start = 0
        x = -1
        y = -1
        for i in range(N):
            if not taken[i]:
                added = True
                ret += 1
                start = i
                taken[i] = True
                x = balls[i][0] + p
                y = balls[i][1] + q
                break
        if not added:
            break
        for i in range(start + 1, N):
            if balls[i][0] == x and balls[i][1] == y:
                taken[i] = True
                x += p
                y += q
    return ret

ans = 100
N = 0
balls = []
taken = []

if __name__ == "__main__":
    I = FastScanner()
    O = OutPut()
    N = I.nextInt()
    ans = N
    balls = [[I.nextInt(), I.nextInt()] for _ in range(N)]
    balls.sort(key=lambda a: a[0])
    solve()
    balls = [[y, x] for x, y in balls]
    balls.sort(key=lambda a: a[0])
    solve()
    O.pln(ans)
# End of Code
